packages/flutter/lib/src/material/dialog.dart

```diff
@@ -961,38 +961,69 @@ Widget _buildMaterialDialogTransitions(BuildContext context, Animation<double> a
 ///  * <https://material.io/design/components/dialogs.html>
 Future<T?> showDialog<T>({
   required BuildContext context,
-  WidgetBuilder? builder,
+  required WidgetBuilder builder,
   bool barrierDismissible = true,
-  Color? barrierColor,
+  Color? barrierColor = Colors.black54,
   bool useSafeArea = true,
   bool useRootNavigator = true,
   RouteSettings? routeSettings,
 }) {
-  assert(builder != null);
```

Comment:
Let's keep this assert for now. (users who have not migrated to NNBD yet can still pass in null)